Userland/Services/WebDriver/Session.cpp

```diff
@@ -297,6 +298,111 @@ ErrorOr<JsonValue, HttpError> Session::get_named_cookie(String const& name)
     return HttpError { 404, "no such cookie", "Cookie not found" };
 }
 
+// POST /session/{session id}/cookie https://w3c.github.io/webdriver/#dfn-adding-a-cookie
+ErrorOr<JsonValue, HttpError> Session::add_cookie(JsonValue const& payload)
+{
+    // 1. Let data be the result of getting a property named cookie from the parameters argument.
+    if (!payload.is_object() || !payload.as_object().has_object("cookie"sv))
+        return HttpError { 400, "invalid argument", "Payload doesn't have a cookie object" };
+
+    auto const& data = payload.as_object().get("cookie"sv).as_object();
```

Comment:
This must be type-checked first (step 2 below), we only know `"cookie"` exists at this point, but it could be any value.